cli/cliui/cliui.go

```diff
@@ -37,22 +41,37 @@ var (
 )
 
 var (
-	Green   = Color("#04B575")
-	Red     = Color("#ED567A")
-	Fuchsia = Color("#EE6FF8")
-	Yellow  = Color("#ECFD65")
-	Blue    = Color("#5000ff")
+	// ANSI color codes
+	red           = Color("1")
+	green         = Color("2")
+	yellow        = Color("3")
+	magenta       = Color("5")
+	white         = Color("7")
+	brightBlue    = Color("12")
+	brightMagenta = Color("13")
 )
 
 // Color returns a color for the given string.
 func Color(s string) termenv.Color {
 	colorOnce.Do(func() {
 		color = termenv.NewOutput(os.Stdout).ColorProfile()
+
 		if flag.Lookup("test.v") != nil {
 			// Use a consistent colorless profile in tests so that results
 			// are deterministic.
 			color = termenv.Ascii
 		}
+
+		// Currently it appears there is no way to use the flag from
+		// serpent as it isn't possible to create a root middleware that
+		// runs for every command. For now we just check if `os.Args`
+		// has the flag.
+		if slices.Contains(os.Args, fmt.Sprintf("--%s", NoColorFlag)) ||
```

Comment:
This is super janky. Relying on os.Args makes it difficult to test. Also, someone could modify the option at the CLI layer and easily forget to update this. Have you considered checking the CLI option higher up in the stack?

Reply:
I absolutely agree with you, however serpent doesn't (from what I can tell) have any way to support this as it lacks the ability to set a middleware that runs before _any_ command.

To demonstrate what I, the code in [this gist](https://gist.github.com/DanielleMaywood/6421e8aa48a160e23d5ddb84093c37c1) has a `Middleware` setup but it only runs on the root cmd and not it's children. If you know of a way to have this run before a commands handler but after options are setup I'd love to do that instead!

Reply:
Scratch that: I think using `rootCmd.Walk` and injecting the middleware to every child should work? Will have a go at doing that.

Reply:
I've pushed a change that should be less janky, let me know what you think!

Reply:
If you do the Walk approach we have the flag duplicated on every help, which is obnoxious. Can't you use an Option on the root command?